src/models/time.js

```diff
@@ -132,6 +133,54 @@ const TimeModel = DataConnected.extend({
     }
   },
 
+  setLinkWith(hook) {
+    this.hooksToListen.add(hook);
+    hook.on("startLoading", () => this.setReady(false));
+    hook.on("ready", this.checkTimeLimits.bind(this));
+  },
+
+  unsetLinkWith(hook) {
+    this.hooksToListen.delete(hook);
+    hook.off("startLoading", () => this.setReady(false));
```

Comment:
Did you check if this worked? Your passing in a new function as the second parameter, which wasn't registered before. Just because you repeat the same text (`() => this.setReady(false)`) doesn't mean it's the same function. A new function is created every time `() => this.setReady(false)` is executed.